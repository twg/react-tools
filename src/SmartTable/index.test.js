/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'
import { SmartTable, Table } from './'

const data = [
  { name: 'Scott', power: 'lazers (pew pew)', cool: 'no' },
  { name: 'James', power: 'healing', cool: 'yes' },
  { name: 'Jean', power: 'firebirds', cool: 'yes' },
  { name: 'Hank', power: 'being blue', cool: 'no' },
  { name: 'Xavier', power: 'brains', cool: 'yes' }
]

describe('Table', () => {
  it('matches snapshot', () => {
    let component = renderer.create(<Table data={data} />)

    expect(component.toJSON()).toMatchSnapshot()
  })
})

describe('SmartTable', () => {
  it('matches snapshot', () => {
    let component = renderer.create(<SmartTable data={data} />)

    expect(component.toJSON()).toMatchSnapshot()
  })
})
