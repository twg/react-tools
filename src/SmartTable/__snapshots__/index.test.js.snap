// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmartTable matches snapshot 1`] = `
<table
  className="table"
>
  <thead
    className="tableHead"
  >
    <tr
      className="tableRow  "
    >
      <th
        className="tableHead"
        onClick={[Function]}
      >
        name
      </th>
      <th
        className="tableHead"
        onClick={[Function]}
      >
        power
      </th>
      <th
        className="tableHead"
        onClick={[Function]}
      >
        cool
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      className="tableRow  "
    >
      <td
        className="tableData"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        Scott
      </td>
      <td
        className="tableData"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        lazers (pew pew)
      </td>
      <td
        className="tableData"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        no
      </td>
    </tr>
    <tr
      className="tableRow  "
    >
      <td
        className="tableData"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        James
      </td>
      <td
        className="tableData"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        healing
      </td>
      <td
        className="tableData"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        yes
      </td>
    </tr>
    <tr
      className="tableRow  "
    >
      <td
        className="tableData"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        Jean
      </td>
      <td
        className="tableData"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        firebirds
      </td>
      <td
        className="tableData"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        yes
      </td>
    </tr>
    <tr
      className="tableRow  "
    >
      <td
        className="tableData"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        Hank
      </td>
      <td
        className="tableData"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        being blue
      </td>
      <td
        className="tableData"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        no
      </td>
    </tr>
    <tr
      className="tableRow  "
    >
      <td
        className="tableData"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        Xavier
      </td>
      <td
        className="tableData"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        brains
      </td>
      <td
        className="tableData"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        yes
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Table matches snapshot 1`] = `
<table
  className="table"
>
  <thead
    className="tableHead"
  >
    <tr
      className="tableRow  "
    >
      <td
        className="tableData"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        name
      </td>
      <td
        className="tableData"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        power
      </td>
      <td
        className="tableData"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        cool
      </td>
    </tr>
  </thead>
  <tbody>
    <tr
      className="tableRow  "
    >
      <td
        className="tableData"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        Scott
      </td>
      <td
        className="tableData"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        lazers (pew pew)
      </td>
      <td
        className="tableData"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        no
      </td>
    </tr>
    <tr
      className="tableRow  "
    >
      <td
        className="tableData"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        James
      </td>
      <td
        className="tableData"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        healing
      </td>
      <td
        className="tableData"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        yes
      </td>
    </tr>
    <tr
      className="tableRow  "
    >
      <td
        className="tableData"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        Jean
      </td>
      <td
        className="tableData"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        firebirds
      </td>
      <td
        className="tableData"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        yes
      </td>
    </tr>
    <tr
      className="tableRow  "
    >
      <td
        className="tableData"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        Hank
      </td>
      <td
        className="tableData"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        being blue
      </td>
      <td
        className="tableData"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        no
      </td>
    </tr>
    <tr
      className="tableRow  "
    >
      <td
        className="tableData"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        Xavier
      </td>
      <td
        className="tableData"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        brains
      </td>
      <td
        className="tableData"
        style={
          Object {
            "width": undefined,
          }
        }
      >
        yes
      </td>
    </tr>
  </tbody>
</table>
`;
